modules/lib/src/main/resources/assets/js/app/dialog/DialogErrorsStateBar.ts

```diff
@@ -0,0 +1,152 @@
+import * as Q from 'q';
+import {i18n} from '@enonic/lib-admin-ui/util/Messages';
+import {DivEl} from '@enonic/lib-admin-ui/dom/DivEl';
+import {DialogErrorStateEntry, ErrorStateEntryConfig} from './DialogErrorStateEntry';
+
+export interface DialogErrorsStateBarConfig {
+    hideIfResolved?: boolean;
+    failText?: string;
+    resolvedText?: string;
+}
+
+enum Modifiers {
+    FAILED = 'failed',
+    CHECKING = 'checking',
+    RESOLVED = 'resolved',
+    HIDDEN = 'hidden',
+}
+
+export type ResolvedStateChangeHandler = (checking: boolean) => void;
+
+export class DialogErrorsStateBar
+    extends DivEl {
+
+    private failEntry: DialogErrorStateEntry;
+
+    private checkEntry: DialogErrorStateEntry;
+
+    private errorsEntries: DialogErrorStateEntry[];
+
+    private resolvedEntry: DialogErrorStateEntry | undefined;
+
+    private hideIfResolved: boolean;
+
+    private readonly resolvedStateChangeHandlers: ResolvedStateChangeHandler[];
+
+    private readonly config: DialogErrorsStateBarConfig;
+
+    constructor(config?: DialogErrorsStateBarConfig) {
+        super('dialog-errors-state-bar');
+
+        this.config = config ?? {};
+
+        this.hideIfResolved = config?.hideIfResolved ?? false;
+
+        this.resolvedStateChangeHandlers = [];
+
+        this.initElements();
+
+        this.updateChecking();
+    }
+
+    doRender(): Q.Promise<boolean> {
+        return super.doRender().then((rendered: boolean) => {
+            this.appendChildren(this.failEntry, this.checkEntry, this.resolvedEntry, ...this.errorsEntries);
+            return rendered;
+        });
+    }
+
+    protected initElements(): void {
+        const {
+            failText = i18n('dialog.errors.fail'),
+            resolvedText = i18n('dialog.errors.resolved'),
+        } = this.config;
+
+        this.failEntry = new DialogErrorStateEntry({
+            className: 'fail-entry',
+            text: failText,
+        });
+
+        this.checkEntry = new DialogErrorStateEntry({
+            className: 'check-entry',
+            text: i18n('dialog.errors.checking'),
+            iconClass: 'icon-spinner',
+        });
+
+        this.errorsEntries = [];
+
+        this.resolvedEntry = new DialogErrorStateEntry({
+            className: 'resolved-entry',
+            text: resolvedText,
+            iconClass: 'icon-checkmark',
+        });
+
+        this.updateResolvedState();
+    }
+
+    protected updateResolvedState(): void {
+        const isResolved = !this.errorsEntries.some(entry => entry.isActive());
```

Comment:
`const isResolved: boolean = !this.errorsEntries.some((entry: DialogErrorStateEntry) => entry.isActive());`